src/Provider/PaymentTokenProvider.php

```diff
@@ -15,26 +21,34 @@ final class PaymentTokenProvider implements PaymentTokenProviderInterface
     public function __construct(
         private readonly RepositoryInterface $orderRepository,
         private readonly RepositoryInterface $paymentTokenRepository,
-    ) {}
+    ) {
+    }
 
     public function provideToken(Request $request): ?PaymentSecurityTokenInterface
     {
+        /** @var string $content */
         $content = $request->getContent();
         $content = json_decode($content, true);
 
         $transactionData = $content['transaction'];
 
         $order = $this->getOrder($transactionData);
 
-        foreach ($order->getPayments() as $payment) {
+        if ($order == null) {
+            return null;
+        }
+
+        $payments = $order->getPayments();
+
+        foreach ($payments as $payment) {
```

Comment:
We can't change the logic here. Please ignore this in phpstan, if it raises an issue here.